### ISSUE TITLE
My price list implementation is no longer working

### ISSUE THREAD

**Issue by BorisKamp** [medusajs/medusa#5261]
Using Medusa `1.17.0`.

I implemented a middleware solution a couple of medusa version back, after Adrien's advice. I have the following middleware that sets the user when calling product endpoints, so I can get the right prices (based on price list):
```javascript
export function setCustomerMiddleware(req, res, next) {
  console.dir(req)
  if (req.get('medusa-customer-id')) {
    req.user = {
      customer_id: req.get('medusa-customer-id')
    }
  }

  next()
}
```
That's my `set-customer-middleware.ts` middleware that I apply in my router:
`router.use('/store', setCustomerMiddleware)`

Which works great, well it used to work great, I got the right `calculated_price` back for each variant in the product when I call `{{medusa-host}}/store/products/prod_01H1PXK8DW6NRG9B8RS9RWBKAT?currency_code=eur` for example. But now I don't anymore....

Im missing the prices in the `variants.prices` array as well, there's only one item which is the default price. 
What changed that made this no longer work? I see the `money-amount.js` file changed a bit, did that break it? if so, how can I repair this?

It also does not feel cool to me that this 'just breaks', how can I/we prevent this from happening in other cases in the future?

**Comment by BorisKamp** [comment]
(I use third party authentication, thats Why I need this workaround....)

**Comment by olivermrbl** [comment]
@adrien2p, can you find time to attend to this?

**Comment by adrien2p** [comment]
I ll have a look asap 👍 
One thing I would suggest for next time, don't upgrade your medusa version without testing in a staging/pre prod env to be sure that you get everything you want working.

Also, the middleware should still work from what I can see, and the endpoint has not changed in the latest version.

@BorisKamp would you be able to share the JSON response and the request url you send in both cases with the previous version you was using and the latest version please.

PS: the medusa auth plugin is not yet compatible with the new medusa version, in case you are using it

**Comment by adrien2p** [comment]
I ll have a look to a pr I am thinking of to see if I can find out the issue

**Comment by BorisKamp** [comment]
@adrien2p I was already working on my reply for 10 minutes with detailled responses etc until I noticed one of my price_list's `ends_at` had a value that is in the past. I feel so stupid, I looked over this. It was exactly the price_list I was using to test (the other did not have a `ends_at` value. So silly...

Im sorry for the work I put you trough but it's all good and working. Im am still having unexplainable behaviour in my draft order line item update tho, I created a bug report for that as well (mentioned up here)

Sorry guys!
